server/src/transactions/dto/get-transactions-filter.dto.ts

```diff
@@ -0,0 +1,5 @@
+import { PartialType } from '@nestjs/mapped-types';
+
+import { CreateTransactionDto } from './create-transaction.dto';
+
+export class GetTransactionsDto extends PartialType(CreateTransactionDto) {}
```

Comment:
### Recommended:
It looks like we're not using this filter DTO at the moment. I recommend dropping it if we're not using it. If you remove it, the `Get()` controller method will return all records in this table.

Alternatively, if you want to implement, you should probably create properties for it instead of extending from `CreateTransactionDto`. [Example](https://jasonwhite.xyz/posts/2020/10/19/crud-api-w/-server-side-pagination-with-nestjs-angular-pagination/)

Reply:
but the properties would be the same as the `CreateTransactionDto` no? (except optional, hence [PartialType](https://docs.nestjs.com/techniques/validation#mapped-types))

and i forgot to write a test case for it but we could use the filterDTO, i.e.
```
jamesrogers@jamesrogpremiseMBP-2 web-app % curl -X GET "localhost:3001/api/transactions"     [{"id":5,"status":"IN_PROGRESS","created_date":"2022-01-29T19:11:38.779Z","asset":{"id":3,"title":"thing 2","description":"its also thing","type":"donation","condition":"","quantity":1}},{"id":4,"status":"COMPLETED","created_date":"2022-01-26T01:08:55.808Z","asset":{"id":1,"title":"thing 1","description":"its a thing","type":"donation","condition":"","quantity":1}}]%       
jamesrogers@jamesrogpremiseMBP-2 web-app % curl -X GET "localhost:3001/api/transactions?status=IN_PROGRESS" \
-H "Content-Type: application/json"
[{"id":5,"status":"IN_PROGRESS","created_date":"2022-01-29T19:11:38.779Z","asset":{"id":3,"title":"thing 2","description":"its also thing","type":"donation","condition":"","quantity":1}}]%
jamesrogers@jamesrogpremiseMBP-2 web-app % 
```

Reply:
I actually, just noticed that there is some filtering happening in the service:
https://github.com/Nonprofit-Exchange-Hub/web-app/blob/fe39084b837da3e368914d190ef622c55e7f35c2/server/src/transactions/transactions.service.ts#L22


But in any case, it's best practice to separate the filtering logic (the createDto is playing that role) from the createDto model, otherwise, you will find yourself making the createDto include more props than the client needs to pass for creating and vice-versa. 

Your call, at this point, it's probably not that important.

Reply:
yeah i just added that 😅  wasn't there before
but that's what i had intended, didn't realize it wasn't working before (had assumed it was since we're doing this in other places)

do you mean separated in some other way? to me these already seem separated the way you want createDto for creating and this getDto has optional properties that we can filter by (and add to like if we wanted to paginate and add a limit property or something)